### PR TITLE
[#96232] Auto expire past unstarted reservations

### PR DESCRIPTION
**Pull request by jdzak**
https://pm.tablexi.com/issues/96232

### CODE REVIEW

File: app/support/auto_expire_reservation.rb [hunk 1]
```diff
@@ -15,7 +15,8 @@ def order_details
 
   def purchased_active_order_details
     OrderDetail.purchased_active_reservations
-      .merge(Reservation.relay_in_progress)
+      .joins("INNER JOIN products ON order_details.product_id = products.id")
+      .joins("INNER JOIN relays ON relays.instrument_id = products.id")
```
Comment (jdzak) [inline app/support/auto_expire_reservation.rb]: This filters out reservation only reservations.

Comment (jhanggi) [inline app/support/auto_expire_reservation.rb]: Would `.joins(:products).joins(:relays)` work? Is there a better way we can filter out res-only instruments? This just seems a little too SQL-y for here.

Comment (jdzak) [inline app/support/auto_expire_reservation.rb]: No unfortunately. I tried `.joins(:product).joins(:relay)` and `.joins(product: :relay)`, but it's complaining there's no relay association on product.

```
ActiveRecord::ConfigurationError:
       Association named 'relay' was not found on Product; perhaps you misspelled it?
     # ./app/support/auto_expire_reservation.rb:21:in `purchased_active_order_details'
     # ./app/support/auto_expire_reservation.rb:13:in `order_details'
     # ./app/support/auto_expire_reservation.rb:3:in `perform'
     # ./spec/app_support/auto_expire_reservation_spec.rb:145:in `block (5 levels) in <top (required)>'
     # ./spec/app_support/auto_expire_reservation_spec.rb:145:in `block (4 levels) in <top (required)>'
```

Comment (jhanggi) [inline app/support/auto_expire_reservation.rb]: Would it make sense to create a scope--thinking something like `usuage_based` on products that checks that `control_mechanism != Relay::CONTROL_MECHANISMS[:manual]`?

**Comment by jdzak** [comment]
Ready for re-review.

**Pull request closed without merging**